src/Microsoft.AspNetCore.Mvc.Core/ApiConventionAttribute.cs

```diff
@@ -2,25 +2,79 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Linq;
+using System.Reflection;
 using Microsoft.AspNetCore.Mvc.Core;
-using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Internal;
 
 namespace Microsoft.AspNetCore.Mvc
 {
+    /// <summary>
+    /// API conventions to be applied to an assembly containing MVC controllers or a single controller.
+    /// <para>
+    /// API conventions are used to influence the output of ApiExplorer. 
+    /// </para>
+    /// </summary>
     [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
-    public sealed class ApiConventionAttribute : Attribute, IFilterMetadata
+    public sealed class ApiConventionAttribute : Attribute
```

Comment:
Consider `ApiConventionTypeAttribute`. Since we'll also have an attribute maps to a specific convention by type/method name.